java/arcs/core/storage/util/HoldQueue.kt

```diff
@@ -14,19 +14,24 @@ package arcs.core.storage.util
 import arcs.core.common.Referencable
 import arcs.core.common.ReferenceId
 import arcs.core.crdt.VersionMap
+import arcs.core.storage.Reference
 import arcs.core.storage.util.HoldQueue.Entity
 import arcs.core.storage.util.HoldQueue.Record
 import kotlinx.coroutines.sync.Mutex
 import kotlinx.coroutines.sync.withLock
 
+typealias Dispatcher = suspend ((suspend () -> Unit)) -> Unit
```

Comment:
Should the argument use the `Op` typealias?

```kotlin
suspend (Op) -> Unit
```

Reply:
Yeah... good question. It definitely could. I was keeping the current implementation decoupled from anything else (like OperationQueue/Op), but it might make sense to be a bit more specific about the dispatching interface here, given the intended usage.

Reply:
And this line is ugly :-)